### ISSUE TITLE
Why access-token is verified in oidc supporting reverse-proxy

### ISSUE THREAD

**Issue by akifkhan01** [louketo/louketo-proxy#675]
In handler, why access-token is verified and used instead of id-token itself for OpenID. Shouldn't id-token be verified primarily in OpenIDC.

Comment snippet from code :
`// Flow: once we exchange the authorization code we parse the ID Token; we then check for an access token,
// if an access token is present and we can decode it, we use that as the session token, otherwise we default
// to the ID Token.`

I'm trying to understand the reason for the same here.

**Comment by p53** [comment]
yes id-token should be validated https://issues.redhat.com/browse/KEYCLOAK-8954

**Comment by p53** [comment]
will make pull request for this

**Comment by p53** [comment]
but there will be still problem with verifying access token in middleware, because right now it uses verifytoken method for id token and thus also checks client_id=aud which may/may not be true so it shouldn't be checked by default, but it can't be turned off
